src/stk/molecular/topology_graphs/host_guest/complex.py

```diff
@@ -4,8 +4,49 @@
 
 """
 
+from typing import Tuple
+from ...molecules import BuildingBlock
 from .vertices import HostVertex, GuestVertex
 from ..topology_graph import TopologyGraph, NullOptimizer
+from dataclasses import dataclass
+import warnings
+
+
+@dataclass(frozen=True)
+class Guest:
+    """
+    Holds the data defining the placement of a guest molecule.
+
+    Attributes
+    -------------
+    building_block : :class:`.BuildingBlock`
+        The guest molecule.
+
+    start_vector : :class:`tuple` of :class:`float`, optional
+        A direction vector which gets aligned with `end_vector`.
+
+    end_vector : :class:`tuple` of :class:`float`, optional
+        A direction vector which determines the rotation applied to
+        the `building_block`. A rotation such that `start_vector` is
+        transformed into `end_vector` is applied.
+
+    displacement : :class:`tuple` of :class:`float`, optional
+        The translational offset of the guest.
+
+    """
+
+    building_block: BuildingBlock
+    start_vector: Tuple[float] = (1., 0., 0.)
+    end_vector: Tuple[float] = (1., 0., 0.)
+    displacement: Tuple[float] = (0., 0., 0.)
+
+    def __repr__(self):
```

Comment:
I think you get a nice repr for free when using `dataclass`. If you override this on purpose, or you prefer the custom one you've provided, ignore this comment.